src/olympia/stats/utils.py

```diff
@@ -0,0 +1,69 @@
+from django.conf import settings
+from google.cloud import bigquery
+
+from olympia.constants.applications import FIREFOX
+
+# This is the mapping between the AMO stats `sources` and the BigQuery columns.
+AMO_TO_BIGQUERY_COLUMN_MAPPING = {
+    'apps': 'dau_by_app_version',
+    'locales': 'dau_by_locale',
+    'os': 'dau_by_app_os',
+    'versions': 'dau_by_addon_version',
+}
+
+
+def rows_to_series(rows, filter_by=None):
+    """Transforms BigQuery rows into series items suitable for the rest of the
+    AMO stats logic."""
+    for row in rows:
```

Comment:
what's the data structure of `row`?  Seems like it's both an object (`row.dau`) and a dict (`row[filter_by]`).

Reply:
it is a class (`Row`) which can be accessed using both notations (attributes and dict-style)